src/subscription_manager/cli_command/abstract_syspurpose.py

```diff
@@ -188,6 +188,8 @@ def _get_valid_fields(self):
                 server_response = self.cp.getOwnerSyspurposeValidFields(org_key)
             except connection.RestlibException as rest_err:
                 log.warning("Unable to get list of valid fields using REST API: {rest_err}".format(rest_err=rest_err))
+                mapped_message: str = ExceptionMapper().get_message(rest_err)
+                system_exit(os.EX_USAGE, mapped_message)
```

Comment:
Isn't this exception block used in any possible failure of the REST communication, and not only for an invalid --org? Unless I'm misreading, the implementation for service-level (which is based on `AbstractSyspurposeCommand` with own bits) seems to use `EX_DATAERR` in a similar case.

Even more: `RestlibException` errors for --list use `EX_SOFTWARE` ...

Reply:
It will catch all the REST failures, yes. The question is "What types of errors should only generate a log entry and which should abort the command?" -- receiving RestlibException means that we're not getting requested data, should we continue with the execution as if we did not collect anything and return the empty dict?
I agree with the return code, something else should be used there, it's not usage error.

Reply:
Ah, my note earlier was only about the exit code ;)

You do raise an interesting point though: `_get_valid_fields()` is used in two flows:
1.  when setting/adding a system purpose value: it is indirectly called (see `_are_provided_values_valid()` and `_is_provided_value_valid()`) to check that the value is in the list of valid values
2. when printing the valid values with --list

In case of (1), since we allow adding/settings values even when not valid, all we do is merely printing a warning and continuing. Hence, I think it would not be nice to fail when only querying for valid values. OTOH, if the REST call for valid values says, maybe the REST call to add/set will fail in the same way too...? So maybe a good idea would be to only warn in this case.

In case of (2), we already print a message about it, even if the exit code is still 0. I think it would be good to instead do the usual error handling, since --list lists the valid values, so any failure defeats it.

Reply:
After internal discussion both of these scenarios should abort the command.

Reply:
And for the exit code -- we have no way of checking what kind of error the server is returning. We have to pick either `EX_DATAERR` or `EX_SOFTWARE`.

Reply:
And both seems equally generic and not exactly fitting with the scope... :-(

It seems the vast majority of the RestlibException handlers use `EX_SOFTWARE`, so maybe it would be a better choice for now.

Reply:
ok